# Problema 1

Las matrices $A$ y $B$ representan las ventas en dos años, en millones de dólares, de los tres productos de una empresa en las cuatro regiones en que está disponible. Es decir, $a_{ij}$ y $b_{ij}$ representan el monto de las ventas del producto $i$ en la región $j$ el primer y segundo año, respectivamente:

<br>

\begin{split}
A = \begin{pmatrix}
 2,6 &  4,8 &  1,8 & 0,9 \\
 3,2 &  4,4 &  2,5 & 2,8 \\
 2,4 &  3,6 &  3,8 & 2,5 \\
\end{pmatrix} \text{ } B= \begin{pmatrix}
 3,6 &  2,5 &  3,0 & 2,5 \\
 4,5 &  5,0 &  3,5 & 3,8 \\
 2,9 &  3,0 &  4,6 & 4,0 \\
\end{pmatrix}
\end{split}

<br>

1. Indique el orden de cada matriz.
2. Calcule la matriz $S = A + B$.
3. Calcule la matriz $D = B - A$.
4. Indique el valor e interprete los elementos $s_{24}$ y $d_{24}$.
5. Para el tercer año de funcionamiento, los tres productos ahora están disponibles en seis regiones. Durante este año, las ventas de esta empresa estarán registradas en una matriz $E$. ¿Es posible determinar la matriz $T = A + B + E$? Justifique su respuesta.

1.
* El orden de la Matriz A es: 3X4
* El orden de la Matriz B es: 3X4

In [ ]:
# 2)

import numpy as np

A=np.array([[2.6, 4.8, 1.8, 0.9],
            [3.2, 4.4, 2.5, 2.8],
            [2.4, 3.6, 3.8, 2.5]]) 

B=np.array([[3.6, 2.5, 3.0, 2.5], 
            [4.5, 5.0, 3.5, 3.8], 
            [2.9, 3.0, 4.6, 4.0]])

S=A+B

print(f"La Matriz S es:{S}")

La Matriz S es:[[6.2 7.3 4.8 3.4]
 [7.7 9.4 6.  6.6]
 [5.3 6.6 8.4 6.5]]


In [ ]:
# 3)

import numpy as np

A=np.array([[2.6, 4.8, 1.8, 0.9],
            [3.2, 4.4, 2.5, 2.8],
            [2.4, 3.6, 3.8, 2.5]]) 

B=np.array([[3.6, 2.5, 3.0, 2.5], 
            [4.5, 5.0, 3.5, 3.8], 
            [2.9, 3.0, 4.6, 4.0]])

D=B-A

print(f"La Matriz D es:{D}")



La Matriz D es:[[ 1.  -2.3  1.2  1.6]
 [ 1.3  0.6  1.   1. ]
 [ 0.5 -0.6  0.8  1.5]]


4.- 
S_24=6.6
Interpretación: Dentro de la Matriz S que corresponde a la suma de las dos Matrices de A Y B, el segundo producto en la cuarta región va a tener un valor total en ventas de 6.6 millones de dolares.

D_24=1
Interpretación: Dentro de la Matriz D que corresponde a la resta de las dos Matrice de B menos A, el segundo producto de la cuarta región va a tener un valor total en ventas de 1 millón de dolares.

5.- No, no es posible determinar la Matriz T a traves de una suma de las Matrices A+B+E ya que la ultima mencionada, posee un orden distinto a las dos anteriores.

# Problema 2

Una fábrica con sucursales en Japón y China, produce tres tipos de componentes $j$, para cinco modelos de computador $i$. Los costos de producción mensual, en miles de dólares, de ambas sucursales se representan en las matrices $J$ y $C$, respectivamente:

<br>

\begin{split}
J = \begin{pmatrix}
 100 &  200 &  150  \\
 120 &  210 &  170  \\
 112 &  200 &  131  \\
 124 &  310 &  170  \\
 136 &  156 &  140  \\
\end{pmatrix} \text{ } C = \begin{pmatrix}
 110 &  190 &  160  \\
 130 &  220 &  180  \\
 146 &  190 &  141  \\
 143 &  312 &  163  \\
 137 &  116 &  156  \\
\end{pmatrix}
\end{split}

<br>

1. ¿Cuál es el orden de las matrices?
2. Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.
3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.
4. Un trabajador indica que, en la sucursal japonesa, el costo de producción del producto $3$, modelo $4$ de computador, es más alto que en la sucursal china. ¿Tiene razón? Justifique apropiadamente.

1.

* El orden de la Matriz J es: 5X3
* El orden de la Matriz C es: 5X3

2.

* Elemento J_21: 120 
  Interpretación: El costo de producción total del primer componentes para el segundo modelo de computador es de 120.000 dolares.

* Elemento C_12: 190
  Interpretación: El costo de producción total del segundo componente para el primer modelo de computador es de 190.000 dolares.


In [7]:
import numpy as np

J=np.array([[100, 200, 150],
            [120, 210, 170],
            [112, 200, 131],
            [124, 310, 170],
            [136, 156, 140]])

C=np.array([[110, 190, 160],
            [130, 220, 180],
            [146, 190, 141],
            [143, 312, 163],
            [137, 116, 156]])

G=J+C

# print(G)

costo_total_por_modelo=np.sum(G, axis=1)

print(f"Lista de 5 elementos con el costo total de cada modelo: {costo_total_por_modelo}")

Lista de 5 elementos con el costo total de cada modelo: [ 910 1030  920 1222  841]


4.
Si, tiene razón, ya que el costo de producción de producto 3, modelo 4 de computador, en la sucursal Japonesa es de un total de 190 mil dolares y el mismo producto en la sucursal de China el total de producción es de 163 mil dolares.

# Problema 3

La empresa Beta tiene cinco plantas en Chile, y en cada una de ellas fabrica siete productos. El número de unidades del producto $i$ fabricadas en la planta $j$ en una semana está representado en la matriz $C$ en el elemento $c_{ij}$.

<br>

\begin{split}
C = \begin{pmatrix}
 120 &  200 &  155 &  125  &  230 \\
 155 &  130 &  110 &  160  &  180 \\
 140 &  205 &  150 &  125  &  155 \\
 130 &  120 &  230 &  225  &  125 \\
 135 &  70  &  190 &  175  &  95  \\
 110 &  265 &  180 &  230  &  180 \\
 185 &  85  &  120 &  135  &  150 \\
\end{pmatrix}
\end{split}

<br>

La empresa se expande, creando cinco plantas en Ecuador que producen los mismos siete productos. La producción semanal de esta filial estará representada por una matriz $E$.

1. Si se sabe que los volúmenes de producción semanal de la filial en Ecuador son un $20\%$ menor que en Chile, calcule la matriz $E$ utilizando *Python*.
2. Si representamos la producción semanal total de la empresa Beta, considerando ambas filiales, por la matriz $T$, escriba la expresión que calcula cada elemento de la matriz $T$ usando notación matricial.
3. Calcule la matriz $T$ utilizando *Python*.
4. Interprete los elementos $c_{34}$ y $t_{72}$ en el contexto del problema.

In [1]:
1.

import numpy as np

C=np.array([[120, 200, 155, 125, 230],
 [155, 130, 110, 160, 180],
 [140, 205, 150, 125, 155],
 [130, 120, 230, 225, 125],
 [135, 70, 190, 175, 95],
 [110, 265, 180, 230, 180],
 [185, 85, 120, 135, 150]])

E=C*0.80

print(f"Matriz E: {E}")

Matriz E: [[ 96. 160. 124. 100. 184.]
 [124. 104.  88. 128. 144.]
 [112. 164. 120. 100. 124.]
 [104.  96. 184. 180. 100.]
 [108.  56. 152. 140.  76.]
 [ 88. 212. 144. 184. 144.]
 [148.  68.  96. 108. 120.]]


2.

* Expresión que calcula cada elemento de la Matriz usando notación Matricial: t_ij = c_ij + e_ij

In [ ]:
3.

C=np.array([[120, 200, 155, 125, 230],
 [155, 130, 110, 160, 180],
 [140, 205, 150, 125, 155],
 [130, 120, 230, 225, 125],
 [135, 70, 190, 175, 95],
 [110, 265, 180, 230, 180],
 [185, 85, 120, 135, 150]])

E=C*0.80

T=C+E

print(f"Matriz T: {T}")

Matriz T: [[216. 360. 279. 225. 414.]
 [279. 234. 198. 288. 324.]
 [252. 369. 270. 225. 279.]
 [234. 216. 414. 405. 225.]
 [243. 126. 342. 315. 171.]
 [198. 477. 324. 414. 324.]
 [333. 153. 216. 243. 270.]]


4.
Elemento C_34 = 125
Interpretación: Quiere decir que en la planta N°4 de la empresa en Chile tiene en su producto N°3, un total de 125 unidades.

Elemento T_72 = 153
Interpretación: Quiere decir que dentro de la producción semanal total considerando ambas filiales, tiene dentro de la planta N°2 en su producto N°7, un total de 153 unidades.

# Problema 4

Una empresa que fabrica zapatos tiene dos plantas, una en Valparaíso y la otra en Temuco. La empresa produce zapatos de color negro, blanco y café, tanto para niños, damas como caballeros. La capacidad actual de producción mensual (en miles de pares) para Valparaíso y Temuco, están dadas por las matrices $V$ y $T$, respectivamente. Para ambas matrices, según el orden en que se mencionan, las filas corresponden al color del calzado y las columnas al tipo de persona que utilizará el calzado.

<br>

\begin{split}
V = \begin{pmatrix}
 30 &  34 &  20  \\
 45 &  20 &  16  \\
 14 &  26 &  25  \\
\end{pmatrix} \text{ } T = \begin{pmatrix}
 35 &  30 &  26  \\
 52 &  25 &  18  \\
 23 &  24 &  32  \\
\end{pmatrix}
\end{split}

<br>

1. Determine la producción mensual total de ambas plantas, representándola en una matriz $A$. Utilice *Python* para resolver.
2. Interprete el elemento $a_{21}$ en el contexto del problema.
3. Si para el próximo año se estima que la producción en Valparaíso aumenta en un $30\%$, mientras que la de Temuco se reduce en un $10\%$, represente la nueva producción total en la matriz $B$.
4. Interprete el elemento $b_{13}$ en el contexto del problema.

In [4]:
1.
V = np.array([[30, 34, 20],
              [45, 20, 16],
              [14, 26, 25]])

T = np.array([[35, 30, 26],
              [52, 25, 18],
              [23, 24, 32]])

A = T+V

print(F" La producción mensual de ambas plantas, es decir la Matriz A es: {A}")

 La producción mensual de ambas plantas, es decir la Matriz A es: [[65 64 46]
 [97 45 34]
 [37 50 57]]


2.
Elemento a_21: 97
Interpretación: Dentro de la producción mensual de ambas plantas, la de Valparaiso y Temuco, el total zapatos de color blanco para niños es de 97 mil pares.

In [5]:
3.
V = np.array([[30, 34, 20],
              [45, 20, 16],
              [14, 26, 25]])

T = np.array([[35, 30, 26],
              [52, 25, 18],
              [23, 24, 32]])

V1 = V*1.30
T1 = T*0.90

B = V1+T1

print(f"La producción total de la Matriz B es: {B}")

La producción total de la Matriz B es: [[ 70.5  71.2  49.4]
 [105.3  48.5  37. ]
 [ 38.9  55.4  61.3]]


4.
Elemento b_13: 49.4
Interpretación: Dentro de la produccion total de ambas filiales, luego de ser aplicado un 30% de aumento para la filial de Valparaiso y un decrecimiento de un 10% en la filial de Temuco, los zapatos para niños de color café van a ser de un total de 49.4 miles de pares.

# Problema 5

Genere una matriz $M$ de orden $8 \times 9$ donde cada término de la matriz está definido por:

$$m_{ij} = 8i + 12j$$

y calcule su transpuesta.

In [8]:
1.
M = np.fromfunction(lambda i, j: 8*(i+1) + 12*(j+1), (8, 9))

print(M)

M_transpuesta=M.T

print("--------------------------------------------------------")

print(M_transpuesta)


[[ 20.  32.  44.  56.  68.  80.  92. 104. 116.]
 [ 28.  40.  52.  64.  76.  88. 100. 112. 124.]
 [ 36.  48.  60.  72.  84.  96. 108. 120. 132.]
 [ 44.  56.  68.  80.  92. 104. 116. 128. 140.]
 [ 52.  64.  76.  88. 100. 112. 124. 136. 148.]
 [ 60.  72.  84.  96. 108. 120. 132. 144. 156.]
 [ 68.  80.  92. 104. 116. 128. 140. 152. 164.]
 [ 76.  88. 100. 112. 124. 136. 148. 160. 172.]]
--------------------------------------------------------
[[ 20.  28.  36.  44.  52.  60.  68.  76.]
 [ 32.  40.  48.  56.  64.  72.  80.  88.]
 [ 44.  52.  60.  68.  76.  84.  92. 100.]
 [ 56.  64.  72.  80.  88.  96. 104. 112.]
 [ 68.  76.  84.  92. 100. 108. 116. 124.]
 [ 80.  88.  96. 104. 112. 120. 128. 136.]
 [ 92. 100. 108. 116. 124. 132. 140. 148.]
 [104. 112. 120. 128. 136. 144. 152. 160.]
 [116. 124. 132. 140. 148. 156. 164. 172.]]


# Problema 6

Genere dos matrices:

*   Una matriz $A$ de orden $5 \times 7$, con $a_{ij} = i \cdot j.$
*   Una matriz $B$ de orden $7 \times 5$, con $b_{ij} = 15i - 2j.$

y calcule $C = A^{T} + \dfrac{1}{2}B$.



In [11]:
1
A = np.fromfunction(lambda i, j: (i+1)*(j+1), (5,7))
B = np.fromfunction(lambda i, j: 15*(i+1)-2*(j+1), (7,5))

C = A.T + (1/2)*B

print(f"La Matriz A: {A}")
print(f"La Matriz B: {B}")
print(f"La Matriz C: {C}")

La Matriz A: [[ 1.  2.  3.  4.  5.  6.  7.]
 [ 2.  4.  6.  8. 10. 12. 14.]
 [ 3.  6.  9. 12. 15. 18. 21.]
 [ 4.  8. 12. 16. 20. 24. 28.]
 [ 5. 10. 15. 20. 25. 30. 35.]]
La Matriz B: [[ 13.  11.   9.   7.   5.]
 [ 28.  26.  24.  22.  20.]
 [ 43.  41.  39.  37.  35.]
 [ 58.  56.  54.  52.  50.]
 [ 73.  71.  69.  67.  65.]
 [ 88.  86.  84.  82.  80.]
 [103. 101.  99.  97.  95.]]
La Matriz C: [[ 7.5  7.5  7.5  7.5  7.5]
 [16.  17.  18.  19.  20. ]
 [24.5 26.5 28.5 30.5 32.5]
 [33.  36.  39.  42.  45. ]
 [41.5 45.5 49.5 53.5 57.5]
 [50.  55.  60.  65.  70. ]
 [58.5 64.5 70.5 76.5 82.5]]


# Problema 7

Una red de sensores recoge datos de temperatura ($°C$), humedad (porcentaje) y presión (hPa) en cinco días distintos (de lunes a viernes). Los datos medidos se muestran en la matriz $D$:

<br>

\begin{equation*}
\begin{matrix}
& \begin{matrix}
 & Lu  \ \ &Ma \ \ &Mi \ \ & Ju \ \ &Vi
\end{matrix} \\
&D = \begin{pmatrix}
20    &  28    &  22    &  26     &  24   \\
52    &  57    &  56    &  54     &  58   \\
1.066 &  1.068 &  1.066 &  1.064  &  1.068 \\
\end{pmatrix} & \ \begin{matrix}
Cº\\
\% \\
hPa
\end{matrix}\\
\end{matrix}
\end{equation*}

<br>

Se sabe que por un error de calibración de los sensores, las medidas registradas deberían ser un $5\%$ menores que las registradas.

1. Calcule en una matriz $S$ los valores correctos de temperatura, humedad y presión. Utilice *Python* para calcular.
2. Obtenga la matriz $S^{T}$.
3. Interprete los elementos $s_{24}$ y $s_{31}^{T}$.


In [17]:
1.
D=np.array([[20, 28, 22, 26, 24],
            [52, 57, 56, 54, 58],
            [1066, 1068, 1066, 1064, 1068]])

S=D*0.95

2.
S_T=S.T

print(f"Matriz S: {S}")
print("---------------------------")
print(f"Matriz S^T: {S_T}")

Matriz S: [[  19.     26.6    20.9    24.7    22.8 ]
 [  49.4    54.15   53.2    51.3    55.1 ]
 [1012.7  1014.6  1012.7  1010.8  1014.6 ]]
---------------------------
Matriz S^T: [[  19.     49.4  1012.7 ]
 [  26.6    54.15 1014.6 ]
 [  20.9    53.2  1012.7 ]
 [  24.7    51.3  1010.8 ]
 [  22.8    55.1  1014.6 ]]


3
Elemento s_24: 51.3
Interpretación: El elmento de la fila N°2, columna N°4 de la Matriz S tiene un valor total de 51.3. Es decir que dentro de la red de sensores, luego de la calibración de un 5% menor, registra que el dia Jueves los datos registran un 51.3% de humedad.

Elemento s^T_31: 20.9
Interpretación: El elemento de la fila N°3, columna N°1 de la Matriz transpuesta de S tiene un valor total de 20.09. Es decir que dentro de la red de sensores, luego de la calibración de un 5% menor y tras hacer la transposición de la Matris S, se registra que el día Miercoles los datos registran una temperatura de 20.9C°.

# Problema 8

Una empresa tecnológica vende software y hardware en tres productos: $P_{1}$, $P_{2}$ y $P_{3}$, en cuatro zonas de la capital: norte, sur, este y oeste. Las ventas del mes, en millones de pesos, están registradas en las matrices $S$ y $H$, para software y hardware respectivamente.

<br>

\begin{split}
S = \begin{pmatrix}
 260 &  480 &  180  &  90  \\
 320 &  440 &  250  &  280 \\
 240 &  360 &  380  &  250  \\
\end{pmatrix} \text{ } H = \begin{pmatrix}
 360 &  250 &  300  &  250  \\
 450 &  500 &  350  &  380 \\
 290 &  300 &  460  &  400  \\
\end{pmatrix}
\end{split}

<br>

1. Calcule en la matriz $T$, del total de ventas de la empresa tecnológica.
2. Un trabajador indica que este mes, la sucursal de la zona sur recaudó más por la venta hardware que software, ¿es correcta su afirmación?
3. Debido al black friday se espera que el siguiente mes, las ventas de software aumenten un $7\%$ y las de hardware aumenten un $3\%$ respecto al mes actual. Encuentra en total de ventas esperadas para el siguiente mes y almacénalo en la matriz $M$. Utiliza *Python* para realizar los cálculos.
4. Interpreta los elementos $t_{24}$ y $m_{31}$ en el contexto del problema.
5. Calcula la matriz $M^{T}$ e interpreta el elemento $m_{41}^{T}$ en el contexto del problema.


In [27]:
1.
S=np.array([[260, 480, 180, 90],
            [320, 440, 250, 280],
            [240, 360, 380, 250]])

H=np.array([[360, 250, 300, 250],
            [450, 500, 350, 380],
            [290, 300, 460, 400]])

T=S+H

a=S-H
b=H-S

print(f"Matriz T: {T}")
# print("-------------------")
# print(a)
# print("-------------------")
# print(b)

Matriz T: [[620 730 480 340]
 [770 940 600 660]
 [530 660 840 650]]


2.
Si, es correcta la afirmación del trabajador

In [20]:
3.
S=np.array([[260, 480, 180, 90],
            [320, 440, 250, 280],
            [240, 360, 380, 250]])

H=np.array([[360, 250, 300, 250],
            [450, 500, 350, 380],
            [290, 300, 460, 400]])

S2=S*1.07
H2=H*1.03

M=S2+H2

print(f"Matriz M: {M}")

Matriz M: [[649.  771.1 501.6 353.8]
 [805.9 985.8 628.  691. ]
 [555.5 694.2 880.4 679.5]]


4.
Elemento t_24: 660
Interpretación: Dentro de la matriz T, es decir la suma de las matrices S y H, en la fila N°2, columna N°4, el valor del producto total es de 660 millones de pesos, el producto coresponde a la zona oeste de la capital y se refiere a el producto 2 (P2).

Elemento m_31: 555.5
Interpretación: Luego de que la matriz S tenga un aumento del 7% y de que la matriz H tenga un aumento del 3%, dentro de la suma de estas dos matrices, en la fila N°3, columna N°1, el valor del producto total de este elemento es de 555.5 millones de pesos, este producto corresponde a la zone norte de la capital y se refiere a el producto 3 (P3).



In [28]:
5.
M_T=M.T

print(f"Matriz M^T= {M_T}")
print("------------------")
print('''Elemento M^T_41: 353.8
         Interpretación: La Matriz M que corresponde a la suma de las matrices S y H 
         luego de un aumento del 7% y 3% respectivamente, dentro de su transposición, el elemento de de la fila N°4,
         columna N°1, que corresponde al producto 1 (P1), zona de la capital oeste, tiene un valor total de 353.8 
         millones de pesos.
      ''')

Matriz M^T= [[649.  805.9 555.5]
 [771.1 985.8 694.2]
 [501.6 628.  880.4]
 [353.8 691.  679.5]]
------------------
Elemento M^T_41: 353.8
         Interpretación: La Matriz M que corresponde a la suma de las matrices S y H 
         luego de un aumento del 7% y 3% respectivamente, dentro de su transposición, el elemento de de la fila N°4,
         columna N°1, que corresponde al producto 1 (P1), zona de la capital oeste, tiene un valor total de 353.8 
         millones de pesos.
      
